File: Support/Fuego/Pythia/pythia-0.5/packages/journal/lib/Entry.cc
// -*- C++ -*-
//
//--------------------------------------------------------------------------------
//
//                              Michael A.G. Aivazis
//                       California Institute of Technology
//                       (C) 1998-2003  All Rights Reserved
//
// <LicenseText>
//
//--------------------------------------------------------------------------------
//

#include <portinfo>

#include <map>
#include <vector>
#include <string>

#include "Entry.h"

using namespace journal;

// helpers

static Entry::meta_t initializeDefaults();

// interface
// meta-methods

Entry::~Entry() {}

// static data

Entry::meta_t Entry::_defaults = initializeDefaults();

// helpers
Entry::meta_t initializeDefaults() {
    Entry::meta_t defaults;

    defaults["category"] = "<unknown>";
    defaults["facility"] = "<unknown>";
    defaults["filename"] = "<unknown>";
    defaults["function"] = "<unknown>";
    defaults["line"] = "<unknown>";

    return defaults;
}

// version
// $Id$

// End of file

File: Support/Fuego/Pythia/pythia-0.5/packages/journal/journal/Entry.py
#!/usr/bin/env python
#
#  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#
#                               Michael A.G. Aivazis
#                        California Institute of Technology
#                        (C) 1998-2003 All Rights Reserved
#
#  <LicenseText>
#
#  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#


from builtins import object


class Entry(object):
    def attribute(self, key, value):
        self.meta[key] = value
        return

    def line(self, msg):
        self.text.append(msg)
        return

    def __init__(self):
        self.meta = {}
        self.text = []
        return


# version
__id__ = "$Id$"

#  End of file
